app/models/bookmark.rb

```diff
@@ -24,19 +24,37 @@ def self.auto_delete_preferences
     )
   end
 
+  # TODO (martin) [POLYBOOK] Not relevant once polymorphic bookmarks are implemented.
   validate :unique_per_post_for_user,
     on: [:create, :update],
     if: Proc.new { |b| b.will_save_change_to_post_id? || b.will_save_change_to_user_id? }
 
+  # TODO (martin) [POLYBOOK] Not relevant once polymorphic bookmarks are implemented.
   validate :for_topic_must_use_first_post,
     on: [:create, :update],
     if: Proc.new { |b| b.will_save_change_to_post_id? || b.will_save_change_to_for_topic? }
 
+  validate :unique_per_bookmarkable,
```

Comment:
We will need a validation to ensure that `bookmarkable_id` and `bookmarkable_type` are both present if one is present too.